aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/CredentialStoreActions.kt

```diff
@@ -33,33 +33,26 @@ object CredentialStoreActions : StoreActions {
                     credentialStore.saveCredential(credentials)
                     legacyCredentialStore.deleteCredential()
                 }
-                CredentialStoreEvent(
-                    CredentialStoreEvent.EventType.LoadCredentialStore(
-                        AmplifyCredentialType.DEVICE_METADATA,
-                        null
-                    )
-                )
+                CredentialStoreEvent(CredentialStoreEvent.EventType.LoadCredentialStore(CredentialType.Amplify))
             } catch (error: CredentialStoreError) {
                 CredentialStoreEvent(CredentialStoreEvent.EventType.ThrowError(error))
             }
             logger.verbose("$id Sending event ${evt.type}")
             dispatcher.send(evt)
         }
 
-    override fun clearCredentialStoreAction(amplifyCredentialType: AmplifyCredentialType, username: String?) =
+    override fun clearCredentialStoreAction(credentialType: CredentialType) =
         Action<CredentialStoreEnvironment>("ClearCredentialStore") { id, dispatcher ->
             logger.verbose("$id Starting execution")
             val evt = try {
-                when (amplifyCredentialType) {
-                    AmplifyCredentialType.AMPLIFY_CREDENTIAL -> {
+                when (credentialType) {
+                    CredentialType.Amplify -> {
                         credentialStore.deleteCredential()
                     }
-                    AmplifyCredentialType.DEVICE_METADATA -> {
-                        username?.let {
-                            credentialStore.deleteDeviceKeyCredential(it)
-                        }
+                    is CredentialType.Device -> {
+                        credentialStore.deleteDeviceKeyCredential(credentialType.username ?: "")
                     }
-                    AmplifyCredentialType.ASF_DEVICE_ID -> TODO()
+                    CredentialType.ASF -> TODO()
```

Comment:
This will crash the app if this code branch is called.

Reply:
Yep, we are aware of this. RN ASF is loaded from Shared Preferences and not credentials store. We will migrate ASF data to store in follow up PR.